src/Encamina.Enmarcha.SemanticKernel.Connectors.Document/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -73,4 +74,151 @@ public static IServiceCollection AddDefaultDocumentConnectorProvider(this IServi
+    /// Adds the <see cref="TxtDocumentConnector"/> implementation of <see cref="IEnmarchaDocumentConnector"/> to the specified <see cref="IServiceCollection"/> as a singleton service.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to add services to.</param>
+    /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
+    public static IServiceCollection AddTxtDocumentConnector(this IServiceCollection services)
+    {
+        return services.AddSingleton<IEnmarchaDocumentConnector, TxtDocumentConnector>();
+    }
+
+    /// <summary>
+    /// Adds the <see cref="VttDocumentConnector"/> implementation of <see cref="IEnmarchaDocumentConnector"/> to the specified <see cref="IServiceCollection"/> as a singleton service.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to add services to.</param>
+    /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
+    public static IServiceCollection AddVttDocumentConnector(this IServiceCollection services)
+    {
+        return services.AddSingleton<IEnmarchaDocumentConnector, VttDocumentConnector>();
+    }
+
+    /// <summary>
+    /// Adds the <see cref="CsvTsvDocumentConnector"/> implementation of <see cref="IEnmarchaDocumentConnector"/> to the specified <see cref="IServiceCollection"/> as a singleton service.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to add services to.</param>
+    /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
+    public static IServiceCollection AddCsvTsvDocumentConnector(this IServiceCollection services)
+    {
+        return services.AddSingleton<IEnmarchaDocumentConnector, CsvTsvDocumentConnector>();
+    }
+
+    /// <summary>
+    /// Adds the <see cref="SkVisionImageDocumentConnector"/> implementation of <see cref="IEnmarchaDocumentConnector"/> to the specified <see cref="IServiceCollection"/> as a singleton service.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to add services to.</param>
+    /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
+    public static IServiceCollection AddSkVisionImageDocumentConnector(this IServiceCollection services)
+    {
+        return services.AddSingleton<IEnmarchaDocumentConnector, SkVisionImageDocumentConnector>();
+    }
+
+    /// <summary>
+    /// Adds the default document connectors to the specified <see cref="IServiceCollection"/>.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to add services to.</param>
+    /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
+    public static IServiceCollection AddDefaultDocumentConnectors(this IServiceCollection services)
```

Comment:
In the future, we might consider creating an extension method in Encamina.Enmarcha.DependencyInjection to automate this process. For instance, we could register all implementations of an interface by scanning the assembly, which would make adding new document connectors more streamlined.

Reply:
We can have several connectors (with different strategies) for a document type. If we register them all, we lose control of which ones we are using.

This `AddDefaultDocumentConnectors` function is just to provide some help when updating from the previous implementation.